agent/definitions.py

```diff
@@ -64,7 +99,7 @@ def accept(self, target: Target) -> bool:
         Returns:
             True if the target is valid; otherwise False.
         """
-        pass
+        raise NotImplementedError()
```

Comment:
I think adding raise NotImplementedError() inside an `@abstractmethod` is unnecessary here. The `@abstractmethod`decorator already makes sure that any subclass has to implement the method, and if they don't, Python will raise a TypeError when you try to instantiate the abstract class. So, the NotImplementedError doesn't really add anything extra and can be left out.

Reply:
From what I understand you are saying that it doesnt matter either way, which makes this comment unnecessary.